lib/bolt/util.rb

```diff
@@ -3,32 +3,19 @@
 module Bolt
   module Util
     class << self
-      def read_config_file(path, default_paths = nil, file_name = 'file')
+      def read_yaml_hash(path, file_name)
         require 'yaml'
 
         logger = Logging.logger[self]
-        path_passed = path
-        if path.nil? && default_paths
-          found_default = default_paths.select { |p| File.exist?(p) }
-          if found_default.size > 1
-            logger.warn "Found #{file_name}s at #{found_default.join(', ')}, using the first"
-          end
-          # Use first found, fall back to first default and try to load even if it didn't exist
-          path = found_default.first || default_paths.first
-        end
-
         path = File.expand_path(path)
-        content = File.open(path, "r:UTF-8") { |f| YAML.safe_load(f.read) }
+        content = File.open(path, "r:UTF-8") { |f| YAML.safe_load(f.read) } || {}
+        unless content.is_a?(Hash)
+          raise Bolt::ValidationError, "Invalid type: #{content.class}. #{path} must contain a Hash or be empty."
```

Comment:
This message doesn't match the format of our other errors well. I also think it should be a 
`Bolt::FileError`. Maybe something like:

```rb
raise Bolt::FileError.new("Invalid content for #{file_name} file: #{path} should be a Hash or empty, not #{content.class}", path)
```

Reply:
Oh, dang i just realized i had not even tested that! updated on your suggestion and added a test.

Reply:
https://github.com/puppetlabs/bolt/pull/1573/commits/5a6ed8e93a076cddf4dedbc86b2679fe736c6b07#diff-82fc2c6cdf3061c5c7e36886addcb9b5R95-R106